docs: add a guide for model inputs

crisbeto opened the pull request:
Adds a guide for `model()` inputs for angular.io. Will port this over to angular.dev once it's finalized.

Segdo commented on aio/content/guide/model-inputs.md:
````diff
@@ -0,0 +1,103 @@
+}
+```
+
+This exposes a `studentAge` input and a `studentChange` output on the `StudentDirective`.
````
```suggestion
This exposes a `studentAge` input and a `studentAgeChange` output on the `StudentDirective`.
```

jelbourn reviewed (commented):
Here's a commit with my proposed changes
https://github.com/jelbourn/angular/commit/96b82924fb2cb4a20ff2cfaefb9332b7209024e1

Feel free to push back on anything here; I tried to restructure the guide to de-emphasize the automatic creation of the output and put more emphasis on the support for two way binding, as well as deferring to the `input` doc to cover aliasing and required vs. optional

jelbourn reviewed (approved):
LGTM

MarkTechson commented on aio/content/guide/model-inputs.md:
```diff
@@ -0,0 +1,141 @@
+You can mark a model input as required or provide an alias in the same way as a
+[standard input](/guide/signal-inputs).
+
+Model inputs do not support input transforms.
```
This, to me, is an important point and I would consider highlighting it with a helpful or some other emphasis block.

crisbeto replied:
It's called out again in the "Differences between `model()` and `input()`" section so I'm not sure if we need any more.

MarkTechson reviewed (changes requested):
I left a few small feedback points. I think it's good to go otherwise.

crisbeto commented on aio/content/guide/model-inputs.md:
```diff
@@ -0,0 +1,141 @@
+
+When you declare a model input in a component or directive, Angular automatically creates a
+corresponding [output](/guide/inputs-outputs) for that model. The output's
+name is the model input's name suffixed with "Change".
```
The only way to access it is through the template. I updated the comment to call it out.

crisbeto committed:
docs: add a guide for model inputs

Adds a guide for `model()` inputs for angular.io. Will port this over to angular.dev once it's finalized.

crisbeto commented:
Feedback has been addressed.

MarkTechson reviewed (approved):
LGTM

dylhunn commented:
This PR was merged into the repository by commit 4efcc74dd6f039373894a515e10c5339f2935c88.
